### PR TITLE
Tree- level order traversal in JS 

### LINKED ISSUE #2553: Tree Level order Traversal
- [x] C
- [x] C++
- [ ] C# - @ajaynair710
- [ ] CoffeeScript
- [ ] Dart - @Valdermaut
- [x] Go
- [x] Java
- [x] JS - @ashishnagpal2498
- [x] Kotlin - @sandhyabhan
- [x] Php
- [x] Python
- [ ] Readme - @SimranMakhija7
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by ashishnagpal2498**
Fixes #2553 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Languages Used:
- JavaScript - JS 

#### Files Added:
- Tree_Levelorder_Traversal.js

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Tree_Levelorder_Traversal/Tree_Levelorder_Traversal.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        }
+    }
+    return root;
+};
+
+/* 
+* Output
+*
+* Level Order traversal of tree is 1 2 3 4 5 6 7
+*/
```
Comment (singh-shreya6) [inline Tree_Levelorder_Traversal/Tree_Levelorder_Traversal.js:24]: Please add blank line at end of file.

File: Tree_Levelorder_Traversal/Tree_Levelorder_Traversal.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            let node = nodes.shift()
+            if(nodes.length > 0) node.next = nodes[0];
+            else node.next = null;
+            if(node.left) queue.push(node.left);
+            if(node.right) queue.push(node.right);
+        }
+    }
+    return root;
+};
+
```
Comment (singh-shreya6) [inline Tree_Levelorder_Traversal/Tree_Levelorder_Traversal.js:19]: Please contribute a complete program with calling function and values like main method in most languages.

**Comment by ashishnagpal2498** [comment]
done @singh-shreya6

File: Tree_Levelorder_Traversal/Tree_Levelorder_Traversal.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+// Creates a Tree Node with input value
```
Comment (singh-shreya6) [inline Tree_Levelorder_Traversal/Tree_Levelorder_Traversal.js:1]: A brief description at start of file. and one blank line after description.

File: Tree_Levelorder_Traversal/Tree_Levelorder_Traversal.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+class Node {
+	constructor(value) {
+		this.value = value;
+		this.left = null;
```
Comment (singh-shreya6) [inline Tree_Levelorder_Traversal/Tree_Levelorder_Traversal.js:5]: only 4 space indentation at the start of each block. You have way too many spaces here.

File: Tree_Levelorder_Traversal/Tree_Levelorder_Traversal.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+		this.right = null;
+	}
+}
+
+//Level Order Traversal 
+var traversal = function (root) {
+	if (root === null) return root;
+	if (root.left === null && root.right === null) {
+		console.log(root.value);
```
Comment (singh-shreya6) [inline Tree_Levelorder_Traversal/Tree_Levelorder_Traversal.js:14]: please add only 4 space indentation.

File: Tree_Levelorder_Traversal/Tree_Levelorder_Traversal.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+		return root;
+	}
+	const queue = [];
+	queue.push(root);
+	while (queue.length > 0) {
+		const node = queue.splice(0, 1)[0];
+		console.log(node.value);
```
Comment (singh-shreya6) [inline Tree_Levelorder_Traversal/Tree_Levelorder_Traversal.js:21]: Please add only 4 space indentation.

File: Tree_Levelorder_Traversal/Tree_Levelorder_Traversal.js (hunk 3 above)
Comment (ashishnagpal2498) [reply Tree_Levelorder_Traversal/Tree_Levelorder_Traversal.js:1]: A brief description about , what is tree level order traversal ?

**Comment by singh-shreya6** [comment]
Please squash your commits so that it can be merged @ashishnagpal2498

**Comment by ashishnagpal2498** [comment]
> 
> 
> Please squash your commits so that it can be merged @ashishnagpal2498

Done @singh-shreya6

**Pull request merged**